coordinator.go

```diff
@@ -10,39 +10,37 @@ import (
 	"time"
 )
 
-func NewCoordinator(funcsCtx, compensateFuncsCtx context.Context, saga *Saga, logStore Store, executionID ...string) *ExecutionCoordinator {
-	c := &ExecutionCoordinator{
-		funcsCtx:           funcsCtx,
-		compensateFuncsCtx: compensateFuncsCtx,
-		saga:               saga,
-		logStore:           logStore,
-	}
-	if len(executionID) > 0 {
-		c.ExecutionID = executionID[0]
-	} else {
-		c.ExecutionID = RandString()
-	}
-	return c
-}
-
 type ExecutionCoordinator struct {
 	ExecutionID string
 
 	aborted          bool
 	executionError   error
 	compensateErrors []error
 
-	funcsCtx           context.Context
-	compensateFuncsCtx context.Context
+	ctx context.Context
 
 	saga *Saga
 
 	logStore Store
 }
 
+func NewCoordinator(ctx context.Context, saga *Saga, logStore Store, executionID ...string) *ExecutionCoordinator {
```

Comment:
While I like the idea to have ctx in methods that work with store I'm not sure that removing one of context is the good idea. First context was used to process direct flow when everything happens as expected. But what if user cancels the request, in that case we need to rollback some steps but we can't use the same already canceled context, because we will not be able to commit any rollback transactions.
Could you please clarify how your solution will work in that case?